code/datums/config/_config.dm

```diff
@@ -0,0 +1,126 @@
+#define CONFIG_FLAG_BOOL               BITFLAG(0)
+#define CONFIG_FLAG_NUM                BITFLAG(1)
+#define CONFIG_FLAG_TEXT               BITFLAG(2)
+#define CONFIG_FLAG_LIST               BITFLAG(3)
+#define CONFIG_FLAG_HAS_VALUE          BITFLAG(4)
```

Comment:
Just adding this here, but there's another type of config option that could be handled that's relatively common. Enums. There's a at least one options that accept a string from a list of possible strings which gets turned into a numeric value on parsing "objectives_disabled". It has 3 possible values.
Also, that kind of field is useful in general.

Reply:
It currently is, upstream. But downstream I got at least another one like that. And I could see more use cases for that.